### PR TITLE
Send email for promoted alerts

### PR DESCRIPTION
**Pull request by Robertgervais**

### CODE REVIEW

File: app/controllers/admin/theft_alerts_controller.rb [hunk 1]
```diff
@@ -26,6 +26,10 @@ def update
 
   private
 
+  def send_promoted_alert_email
+    EmailPromotedAlertWorker.new.perform(@theft_alert)
```
Comment (sethherr) [inline app/controllers/admin/theft_alerts_controller.rb]: Needs to be async

File: app/controllers/admin/theft_alerts_controller.rb [hunk 2]
```diff
@@ -60,6 +64,8 @@ def set_alert_timestamps(theft_alert_attrs)
       theft_alert_plan = TheftAlertPlan.find(theft_alert_attrs[:theft_alert_plan_id])
       now = Time.current
       theft_alert_attrs[:begin_at] = now
+      send_promoted_alert_email
+      CustomerMailer.promoted_alert_email(@bike).deliver_now
```
Comment (sethherr) [inline app/controllers/admin/theft_alerts_controller.rb]: Why are you also doing deliver now?

File: app/views/customer_mailer/promoted_alert_email.html.haml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+%h1
+  = @user.name
+%p
+  = t(".html.sorry_to_hear_about_your_bike", brand: @bike.manufacturer)
+%p
+  = t(".html.your_promoted_alert", city: @user.city.present? ? @user.city : "your city", id: @bike.id)
+
+= image_tag @bike.thumb_path
```
Comment (sethherr) [inline app/views/customer_mailer/promoted_alert_email.html.haml]: Should use the promoted alert image.

Should also use the bike box

File: app/mailers/customer_mailer.rb [hunk 4]
```diff
@@ -128,4 +128,15 @@ def updated_terms_email(user)
       mail(to: @user.email, from: '"Lily Williams" <lily@bikeindex.org>')
     end
   end
+
+  def promoted_alert_email(theft_alert)
+    @theft_alert = theft_alert
+    @user = theft_alert.creator
+    @bike = theft_alert.bike
+    @alert_image = @bike&.current_stolen_record&.current_alert_image&.image
```
Comment (sethherr) [inline app/mailers/customer_mailer.rb]: You can use the `alert_image_url` method on bike

File: app/views/customer_mailer/promoted_alert_email.html.haml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+%h1
+  = @user.name
+%p
+  = t(".html.sorry_to_hear_about_your_bike", brand: @bike.manufacturer)
+%p
+  = t(".html.your_promoted_alert", city: @user.city.present? ? @user.city : "your city", id: @bike.id)
```
Comment (sethherr) [inline app/views/customer_mailer/promoted_alert_email.html.haml]: This definitely should be based on the stolen bike, not the user

File: app/views/customer_mailer/promoted_alert_email.text.haml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+
+= @user.name
+
+= t(".html.sorry_to_hear_about_your_bike", brand: @bike.manufacturer)
+
+= t(".html.your_promoted_alert", city: @user.city.present? ? @user.city : "your city", id: @bike.id)
+- if @alert_image.present?
+  = image_tag @alert_image.url
```
Comment (sethherr) [inline app/views/customer_mailer/promoted_alert_email.text.haml]: This is a text email, you can't include html tags

**Comment by Robertgervais** [comment]
Image 8-22-19 at 3 20 PM

File: app/controllers/admin/theft_alerts_controller.rb [hunk 7]
```diff
@@ -26,6 +26,10 @@ def update
 
   private
 
+  def send_promoted_alert_email
```
Comment (sethherr) [inline app/controllers/admin/theft_alerts_controller.rb]: This should be a method on the model

File: app/views/customer_mailer/promoted_alert_email.text.haml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+
+= t(".text.sorry_to_hear_about_your_bike", brand: @bike.manufacturer.name)
+
+= t(".text.your_promoted_alert", city: @theft_alert.stolen_record.city.present? ? @theft_alert.stolen_record.city : "your city", id: @bike.id)
+
+- if @alert_image.present?
+  = image_tag @alert_image
+
+= t(".text.additional_info")
```
Comment (sethherr) [inline app/views/customer_mailer/promoted_alert_email.text.haml]: All these text blocks can be combined into one translation block, for ease.

**Comment by sethherr** [comment]
No longer relevant

**Pull request closed without merging**